Description: removing embedded copies of ezxml, which has many security issues
 and is no more used by scilab anyway.
Author: Pierre Gruet <pgt@debian.org>
Bug-Debian: https://bugs.debian.org/989364
Forwarded: https://gitlab.com/scilab/scilab/-/issues/17527
Last-Update: 2026-01-29

--- a/scilab/modules/scicos/Makefile.am
+++ b/scilab/modules/scicos/Makefile.am
@@ -55,7 +55,6 @@
 src/c/import.c \
 src/c/scicos.c \
 src/c/scicos_free.c \
-src/c/ezxml.c \
 src/c/var2sci.c \
 src/c/MlistGetFieldNumber.c \
 src/c/lsodar.c \
--- a/scilab/modules/scicos/src/c/Scicos_Export.def
+++ b/scilab/modules/scicos/src/c/Scicos_Export.def
@@ -20,5 +20,3 @@
 Coserror
 Set_Jacobian_flag
 Get_Jacobian_parameter
-write_xml_states
-read_xml_initial_states
--- a/scilab/modules/scicos/src/c/cli/scicos-cli_Export.def
+++ b/scilab/modules/scicos/src/c/cli/scicos-cli_Export.def
@@ -20,5 +20,3 @@
 Coserror
 Set_Jacobian_flag
 Get_Jacobian_parameter
-write_xml_states
-read_xml_initial_states
--- a/scilab/modules/scicos_blocks/includes/scicos_block.h
+++ b/scilab/modules/scicos_blocks/includes/scicos_block.h
@@ -92,8 +92,6 @@
 int rhojac_(double *, double *, double *, double *, int *, double *, int *);
 int rho_( double *, double *, double *, double *, double *, int *);
 int fx_( double *, double *);
-int read_xml_initial_states(int nvar, const char * xmlfile, char **ids, double *svars);
-int write_xml_states(int, const char *, char **, double *);
 
 #endif /* __SCICOS_BLOCK_H__ */
 
--- a/scilab/modules/scicos_blocks/includes/scicos_block4.h
+++ b/scilab/modules/scicos_blocks/includes/scicos_block4.h
@@ -122,8 +122,6 @@
 int rhojac_(double *, double *, double *, double *, int *, double *, int *);
 int rho_(double *, double *, double *, double *, double *, int *);
 int fx_(double *, double *);
-int read_xml_initial_states(int nvar, const char *xmlfile, char **ids, double *svars);
-int write_xml_states(int, const char *, char **, double *);
 
 /** Define scicos simulator data type number (_N) */
 typedef enum
--- a/scilab/modules/scicos/src/c/scicos.c
+++ b/scilab/modules/scicos/src/c/scicos.c
@@ -69,7 +69,6 @@
 #include "syncexec.h"
 #include "realtime.h"
 #include "sci_malloc.h"
-#include "ezxml.h"
 
 #include "sciblk2.h"
 #include "sciblk4.h"
@@ -287,7 +286,6 @@
 static void addevs(double t, int *evtnb, int *ierr1);
 static int synchro_g_nev(ScicosImport *scs_imp, double *g, int kf, int *ierr);
 static void Multp(double *A, double *B, double *R, int ra, int rb, int ca, int cb);
-static int read_id(ezxml_t *elements, char *id, double *value);
 static int simblkdaskr(realtype tres, N_Vector yy, N_Vector yp, N_Vector resval, void *rdata);
 static void SundialsErrHandler(int error_code, const char *module, const char *function, char *msg, void *user_data);
 static int Jacobians(realtype tt, realtype cj, N_Vector yy,
@@ -6593,95 +6591,6 @@
     return;
 }
 /*--------------------------------------------------------------------------*/
-int read_xml_initial_states(int nvar, const char * xmlfile, char **ids, double *svars)
-{
-    ezxml_t model, elements;
-    int result = 0, i = 0;
-    double vr = 0.;
-
-    if (nvar == 0)
-    {
-        return 0;
-    }
-    result = 0;
-    for (i = 0; i < nvar; i++)
-    {
-        if (strcmp(ids[i], "") != 0)
-        {
-            result = 1;
-            break;
-        }
-    }
-    if (result == 0)
-    {
-        return 0;
-    }
-
-    model = ezxml_parse_file(xmlfile);
-
-    if (model == NULL)
-    {
-        sciprint(_("Error: Cannot find file '%s'.\n"), xmlfile);
-        return -1;/* file does not exist*/
-    }
-
-    elements = ezxml_child(model, "elements");
-    for (i = 0; i < nvar; i++)
-    {
-        vr = 0.0;
-        result = read_id(&elements, ids[i], &vr);
-        if (result == 1)
-        {
-            svars[i] = vr;
-        }
-    }
-    ezxml_free(model);
-    return 0;
-}
-/*--------------------------------------------------------------------------*/
-static int read_id(ezxml_t *elements, char *id, double *value)
-{
-    char V1[100], V2[100];
-    int ok = 0, i = 0, ln = 0;
-
-    if (strcmp(id, "") == 0)
-    {
-        return 0;
-    }
-    ok = search_in_child(elements, id, V1);
-    if (ok == 0 )
-    {
-        /*sciprint(_("Cannot find: %s=%s  \n"),id,V1);      */
-        return 0;
-    }
-    else
-    {
-        if (Convert_number(V1, value) != 0)
-        {
-            ln = (int)(strlen(V1));
-            if (ln > 2)
-            {
-                for (i = 1; i <= ln - 2; i++)
-                {
-                    V2[i - 1] = V1[i];
-                }
-                V2[ln - 2] = '\0';
-                ok = read_id(elements, V2, value);
-                return ok;
-            }
-            else
-            {
-                return 0;
-            }
-        }
-        else
-        {
-            /*      printf("\n ---->>>%s= %g",V1,*value);*/
-            return 1;
-        }
-    }
-}
-/*--------------------------------------------------------------------------*/
 int Convert_number(char *s, double *out)
 {
     char *endp = NULL;
@@ -6711,88 +6620,6 @@
     }
 }
 /*--------------------------------------------------------------------------*/
-int write_xml_states(int nvar, const char * xmlfile, char **ids, double *x)
-{
-    ezxml_t model, elements;
-    int result = 0, i = 0, err = 0;
-    FILE *fd = NULL;
-    char *s = NULL;
-    char **xv = NULL;
-
-    if (nvar == 0)
-    {
-        return 0;
-    }
-    result = 0;
-    for (i = 0; i < nvar; i++)
-    {
-        if (strcmp(ids[i], "") != 0)
-        {
-            result = 1;
-            break;
-        }
-    }
-    if (result == 0)
-    {
-        return 0;
-    }
-
-    xv = MALLOC(nvar * sizeof(char*));
-    for (i = 0; i < nvar; i++)
-    {
-        xv[i] = MALLOC(nvar * 100 * sizeof(char));
-        sprintf(xv[i], "%g", x[i]);
-    }
-
-    model = ezxml_parse_file(xmlfile);
-    if (model == NULL)
-    {
-        sciprint(_("Error: Cannot find file '%s'.\n"), xmlfile);
-        err = -1;
-        goto err_free_xv;
-    }
-
-    elements = ezxml_child(model, "elements");
-
-    for (i = 0; i < nvar; i++)
-    {
-        if (strcmp(ids[i], "") == 0)
-        {
-            continue;
-        }
-        result = write_in_child(&elements, ids[i], xv[i]);
-        if (result == 0 )
-        {
-            /* sciprint(_("cannot find %s in '%s' \n"),ids[i],xmlfile);      */
-            /* err= -1;*/ /* Variable does not exist*/
-        }
-    }
-
-    s = ezxml_toxml(model);
-    ezxml_free(model);
-
-
-    wcfopen(fd, (char*)xmlfile, "wb");
-    if (fd == NULL)
-    {
-        err = -3;/* cannot write to file*/
-        goto err_free_s;
-    }
-
-    fputs (s, fd);
-    fclose(fd);
-
-err_free_s:
-    free(s);
-err_free_xv:
-    for (i = 0; i < nvar; i++)
-    {
-        FREE(xv[i]);
-    }
-    FREE(xv);
-    return err;
-}
-/*--------------------------------------------------------------------------*/
 int C2F(fx)(double *x, double *residual) /* used for homotopy*/
 {
     double  *xdot = NULL, t = 0;
